// ng-terminal/esm5/lib/global-style/global-style.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/global-style/global-style.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ViewEncapsulation } from '@angular/core';
var GlobalStyleComponent = /** @class */ (function () {
    function GlobalStyleComponent() {
    }
    /**
     * @return {?}
     */
    GlobalStyleComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    GlobalStyleComponent.decorators = [
        { type: Component, args: [{
                    selector: 'global-style',
                    template: "",
                    //global styles
                    encapsulation: ViewEncapsulation.None,
                    styles: [".xterm{position:relative;-moz-user-select:none;user-select:none;-ms-user-select:none;-webkit-user-select:none;cursor:text}.xterm.focus,.xterm:focus{outline:0}.xterm .xterm-helpers{position:absolute;top:0;z-index:5}.xterm .xterm-helper-textarea{padding:0;border:0;margin:0;position:absolute;opacity:0;left:-9999em;top:0;width:0;height:0;z-index:-5;white-space:nowrap;overflow:hidden;resize:none}.xterm .composition-view{background:#000;color:#fff;display:none;position:absolute;white-space:nowrap;z-index:1}.xterm .composition-view.active{display:block}.xterm .xterm-viewport{background-color:#000;overflow-y:scroll;cursor:default;position:absolute;right:0;left:0;top:0;bottom:0}.xterm .xterm-screen{position:relative}.xterm .xterm-screen canvas{position:absolute;left:0;top:0}.xterm .xterm-scroll-area{visibility:hidden}.xterm-char-measure-element{display:inline-block;visibility:hidden;position:absolute;top:0;left:-9999em;line-height:normal}.xterm.enable-mouse-events{cursor:default}.xterm.xterm-cursor-pointer{cursor:pointer}.xterm.column-select.focus{cursor:crosshair}.xterm .xterm-accessibility,.xterm .xterm-message{position:absolute;left:0;top:0;bottom:0;right:0;z-index:10;color:transparent}.xterm .live-region{position:absolute;left:-9999px;width:1px;height:1px;overflow:hidden}.xterm-dim{opacity:.5}.xterm-underline{text-decoration:underline}.xterm-strikethrough{text-decoration:line-through}"]
                }] }
    ];
    /** @nocollapse */
    GlobalStyleComponent.ctorParameters = function () { return []; };
    return GlobalStyleComponent;
}());
export { GlobalStyleComponent };
