// survey-creator-angular/esm2015/components/action-button.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "survey-angular-ui";
export class ActionButtonComponent extends EmbeddedViewContentComponent {
    constructor() {
        super(...arguments);
        this.classes = "";
        this.selected = false;
        this.disabled = false;
        this.text = "";
        this.title = "";
        this.allowBubble = false;
    }
    onClick(event) {
        this.click();
        if (!this.allowBubble) {
            event.stopPropagation();
        }
    }
}
ActionButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ActionButtonComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ActionButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ActionButtonComponent, selector: "svc-action-button", inputs: { classes: "classes", click: "click", selected: "selected", disabled: "disabled", text: "text", title: "title", allowBubble: "allowBubble" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <span *ngIf=\"disabled\" class=\"svc-action-button svc-action-button--disabled\"\n    [class]=\"classes\" [attr.title]=\"title\">{{text}}</span>\n  <span *ngIf=\"!disabled\" role=\"button\" class=\"svc-action-button\"\n    (click)=\"onClick($event)\" [key2click] [attr.title]=\"title\" [class.svc-action-button--selected]=\"selected\" [class]=\"classes\">{{text}}</span>\n</ng-template>", styles: [":host { display: none; }"], directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.Key2ClickDirective, selector: "[key2click]", inputs: ["key2click"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ActionButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-action-button",
                    templateUrl: "./action-button.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { classes: [{
                type: Input
            }], click: [{
                type: Input
            }], selected: [{
                type: Input
            }], disabled: [{
                type: Input
            }], text: [{
                type: Input
            }], title: [{
                type: Input
            }], allowBubble: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-action-button", ActionButtonComponent);
