// survey-creator-angular/esm2020/creator-model.component.mjs
import { Component } from "@angular/core";
import { BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
export class CreatorModelComponent extends BaseAngular {
    createHash() {
        this.hash = {};
        this.getPropertiesToTrack().forEach((prop) => {
            this.hash[prop] = this[prop];
        });
    }
    ngOnInit() {
        this.createModel();
        this.createHash();
        super.ngOnInit();
    }
    ngDoCheck() {
        if (this.needUpdateModel())
            this.createModel();
        super.ngDoCheck();
    }
    needUpdateModel() {
        let res = false;
        Object.keys(this.hash).forEach(key => {
            if (this.hash[key] != this[key]) {
                this.hash[key] = this[key];
                res = true;
            }
        });
        return res;
    }
}
CreatorModelComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CreatorModelComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
CreatorModelComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: CreatorModelComponent, selector: "ng-component", usesInheritance: true, ngImport: i0, template: "", isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CreatorModelComponent, decorators: [{
            type: Component,
            args: [{
                    template: ""
                }]
        }] });
