// survey-creator-angular/esm2020/row.component.mjs
import { Component, Input } from "@angular/core";
import { RowViewModel } from "survey-creator-core";
import { CreatorModelComponent } from "./creator-model.component";
import { AngularComponentFactory } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
export class CreatorRowComponent extends CreatorModelComponent {
    get row() {
        return this.componentData.row;
    }
    get creator() {
        return this.componentData.creator;
    }
    getModel() {
        return this.model;
    }
    createModel() {
        this.model = new RowViewModel(this.creator, this.row, undefined);
    }
    getPropertiesToTrack() {
        return ["creator", "row"];
    }
}
CreatorRowComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CreatorRowComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
CreatorRowComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: CreatorRowComponent, selector: "svc-row", inputs: { componentData: "componentData" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"model.cssClasses\">\n    <sv-ng-row [row]=\"row\"></sv-ng-row>\n  </div>\n</ng-template>", components: [{ type: i1.RowComponent, selector: "sv-ng-row", inputs: ["row"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CreatorRowComponent, decorators: [{
            type: Component,
            args: [{ selector: "svc-row", template: "<ng-template #template>\n  <div [class]=\"model.cssClasses\">\n    <sv-ng-row [row]=\"row\"></sv-ng-row>\n  </div>\n</ng-template>" }]
        }], propDecorators: { componentData: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-row", CreatorRowComponent);
